### ISSUE TITLE
Source code for route file getting modified by the CLI

### ISSUE THREAD

**Issue by universse** [TanStack/router#883]
### Describe the bug

In my app I'm using double quote by default. Running `tsr watch`/`tsr generate` will update the code to use single quote. Is there a way to work around this?

```js
export const Route = new FileRoute("/")
```

becomes

```js
export const Route = new FileRoute('/')
```

### Your Example Website or App

any of the file-base routing examples

### Steps to Reproduce the Bug or Issue

NA

### Expected behavior

NA

### Platform

NA

**Comment by schiller-manuel** [comment]
Currently it's hardcoded:

https://github.com/TanStack/router/blob/beta/packages/router-cli/src/generator.ts#L186-L189

But could be made configurable...

**Comment by bswan0002** [comment]
A very easy solution for this use-case would be to use prettier and configure a script to run prettier immediately following `tsr generate` in `package.json`:

```json
"scripts": {
  "generate-and-format": "tsr generate && prettier --write 'src/[yourRoutesDirectory]/*.{js,jsx,ts,tsx}'"
}
```

Then any time you'd run `tsr generate` you can instead run `npm run generate-and-format` to achieve the desired formatting.

**Comment by universse** [comment]
why is it necessary to pass `path` to `FileRoute` when `path` is computed from the file path?

**Comment by tannerlinsley** [comment]
TypeScript has no idea what file you’re in, or at least doesn’t expose it for use as a type.

**Comment by universse** [comment]
oh I was under the impression that the type inference would come from the generated routeTree file, rather than the path that gets passed to `FileRoute`.

**Comment by tannerlinsley** [comment]
It needs both. The path in the type is the key to getting the routes parent types.

**Comment by schiller-manuel** [comment]
#894 (released in v1.0.6) adds a configuration option for the quote style.

**Comment by universse** [comment]
appreciate your fast response on this
